firebase-components/src/main/java/com/google/firebase/components/ComponentRuntime.java

```diff
@@ -122,16 +125,41 @@ private void discoverComponents(List<Component<?>> componentsToAdd) {
           Log.w(ComponentDiscovery.TAG, "Invalid component registrar.", ex);
         }
       }
+      for (int i = 0; i < processedComponents.size(); i++) {
+        Object[] interfaces = processedComponents.get(i).getProvidedInterfaces().toArray();
+        for (Object anInterface : interfaces) {
+          if (anInterface.toString().contains("kotlinx.coroutines.CoroutineDispatcher")) {
+            existingInterfaces.add(anInterface.toString());
+          }
+        }
+      }
+      for (int i = 0; i < componentsToAdd.size(); i++) {
```

Comment:
Do you have any idea how expensive these loops can be? Since this code runs during initialization and, AFAIR in the main thread, any extra work will impact start up for all apps.

An alternative would be to set a class field boolean that works as a flag of whether we have seen the dispatcher before. Then, we would only need to loop once, and we wouldn't need to create an additional list and copy over every component (`componentsAdding`) nor the list of interfaces already seen (`existingInterfaces`)

WDYT?